File: app/src/main/java/com/example/smoiapp001/utilities/TransactionUtils.java
package com.example.smoiapp001.utilities;

import com.example.smoiapp001.database.models.TransactionEntry;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.List;

public class TransactionUtils {

    // Constant for editing transaction period, 1 day
    private static final long EDITABLE_TRANSACTION_PERIOD = DateUtils.DAY_IN_MILLISECOND;

    public static ArrayList<String> getDescriptionList(List<TransactionEntry> transactions) {
        ArrayList<String> descriptions = new ArrayList<>();
        for (TransactionEntry transaction : transactions) {
            String newDescription = transaction.getDescription();
            if (!descriptions.contains(newDescription)) {
                descriptions.add(newDescription);
            }
        }
        return descriptions;
    }

    public static ArrayList<TransactionEntry> getTransactionByDate(List<TransactionEntry> transactions,
                                                                   Calendar calendar) {
        ArrayList<TransactionEntry> selectedTransactions = new ArrayList<>();
        for (TransactionEntry transaction : transactions) {
            Calendar entryCalendar = Calendar.getInstance();
            entryCalendar.setTime(transaction.getDate());

            if (calendar.get(Calendar.DAY_OF_YEAR) == entryCalendar.get(Calendar.DAY_OF_YEAR)) {
                selectedTransactions.add(transaction);
            }
        }
        return selectedTransactions;
    }

    public static float calculateDayCost(List<TransactionEntry> transactions,
                                         Calendar calendar) {
        float daySum = 0;
        for (TransactionEntry transaction : transactions) {
            Calendar entryCalendar = Calendar.getInstance();
            entryCalendar.setTime(transaction.getDate());

            if (calendar.get(Calendar.YEAR) == entryCalendar.get(Calendar.YEAR) &&
                    calendar.get(Calendar.MONTH) == entryCalendar.get(Calendar.MONTH) &&
                    calendar.get(Calendar.DAY_OF_MONTH) == entryCalendar.get(Calendar.DAY_OF_MONTH)) {
                daySum += transaction.getCost();
            }

        }
        return daySum;
    }

    public static float calculateMonthCost(List<TransactionEntry> transactions,
                                         Calendar calendar) {
        float monthSum = 0;
        for (TransactionEntry transaction : transactions) {
            Calendar entryCalendar = Calendar.getInstance();
            entryCalendar.setTime(transaction.getDate());

            if (calendar.get(Calendar.YEAR) == entryCalendar.get(Calendar.YEAR) &&
                    calendar.get(Calendar.MONTH) == entryCalendar.get(Calendar.MONTH)) {
                monthSum += transaction.getCost();
            }

        }
        return monthSum;
    }

    public static float getCostModeByDescription(String description) {

        return 0;
    }

    public static String[] getDescriptionSearchResult(String description) {

        return null;
    }

    public static boolean isLocked(TransactionEntry transaction) {
        long nowTimestamp = DateUtils.toTimestamp(new Date());
        long transactionTimestamp = DateUtils.toTimestamp(transaction.getDate());
        if (nowTimestamp - transactionTimestamp >= EDITABLE_TRANSACTION_PERIOD) {
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/example/smoiapp001/utilities/TransactionUtils.kt
package com.example.smoiapp001.utilities

import com.example.smoiapp001.database.models.TransactionEntry
import java.util.ArrayList
import java.util.Calendar
import java.util.Date

object TransactionUtils {

    // Constant for editing transaction period, 1 day
    private val EDITABLE_TRANSACTION_PERIOD = DateUtils.DAY_IN_MILLISECOND

    fun getDescriptionList(transactions: List<TransactionEntry>): ArrayList<String> {
        val descriptions = ArrayList<String>()
        for (transaction in transactions) {
            val newDescription = transaction.description
            if (!descriptions.contains(newDescription)) {
                descriptions.add(newDescription!!)
            }
        }
        return descriptions
    }

    fun getTransactionByDate(transactions: List<TransactionEntry>,
                             calendar: Calendar): ArrayList<TransactionEntry> {
        val selectedTransactions = ArrayList<TransactionEntry>()
        for (transaction in transactions) {
            val entryCalendar = Calendar.getInstance()
            entryCalendar.time = transaction.date

            if (calendar.get(Calendar.DAY_OF_YEAR) == entryCalendar.get(Calendar.DAY_OF_YEAR)) {
                selectedTransactions.add(transaction)
            }
        }
        return selectedTransactions
    }

    fun calculateDayCost(transactions: List<TransactionEntry>,
                         calendar: Calendar): Float {
        var daySum = 0f
        for (transaction in transactions) {
            val entryCalendar = Calendar.getInstance()
            entryCalendar.time = transaction.date

            if (calendar.get(Calendar.YEAR) == entryCalendar.get(Calendar.YEAR) &&
                    calendar.get(Calendar.MONTH) == entryCalendar.get(Calendar.MONTH) &&
                    calendar.get(Calendar.DAY_OF_MONTH) == entryCalendar.get(Calendar.DAY_OF_MONTH)) {
                daySum += transaction.cost
            }

        }
        return daySum
    }

    fun calculateMonthCost(transactions: List<TransactionEntry>,
                           calendar: Calendar): Float {
        var monthSum = 0f
        for (transaction in transactions) {
            val entryCalendar = Calendar.getInstance()
            entryCalendar.time = transaction.date

            if (calendar.get(Calendar.YEAR) == entryCalendar.get(Calendar.YEAR) && calendar.get(Calendar.MONTH) == entryCalendar.get(Calendar.MONTH)) {
                monthSum += transaction.cost
            }

        }
        return monthSum
    }

    fun getCostModeByDescription(description: String): Float {

        return 0f
    }

    fun getDescriptionSearchResult(description: String): Array<String>? {

        return null
    }

    fun isLocked(transaction: TransactionEntry): Boolean {
        val nowTimestamp = DateUtils.toTimestamp(Date())
        val transactionTimestamp = DateUtils.toTimestamp(transaction.date!!)
        return nowTimestamp - transactionTimestamp >= EDITABLE_TRANSACTION_PERIOD
    }
}
